### DISCUSSION TITLE
Can't open a file after downloading it.

### DISCUSSION THREAD

**Discussion by fabiofilipegomes** [capawesome-team/capacitor-plugins#177] [Q&A] [answered]
Hi,

I'm trying to open a file after downloading and storing it for offline purposes.
Here's my code below.

Download: 
`let hasPermissions = await this.getFilesystemAccess();
            if(!hasPermissions) {
                resolve(false);
                return;
            }

            this.checkFileExists(environment.mediaPath, mediaGuid.toString(), extension).then(exists => {
                if (!exists) {
                    Filesystem.downloadFile({
                        path: (environment.mediaPath + mediaGuid + "." + extension),
                        url: environment.mediaApiUrl + mediaGuid,
                        directory: Directory.Data
                    }).then(x => {
                        this.errorCtrl.presentToast("downloaded?:" + (x.path ? x.path : "nope"));
                    }).catch(x => {
                        this.errorCtrl.presentToast("downloaded err:" + JSON.stringify(x));
                    });
                    resolve(true);
                } else {
                    resolve(exists ? exists : false);
                }
            });`

Open file:

`let fileUri = await Filesystem.getUri({
                    directory: Directory.Data,
                    path: environment.mediaPath + document.guid + "." + ext
                });
                await FileOpener.openFile({ path: fileUri.uri })
                    .catch(error => this.errorCtrl.presentToast("error:" + JSON.stringify(error) + " - " + error.toString()));`

Some print screens attached as well.
![Screenshot_1](https://github.com/capawesome-team/capacitor-plugins/assets/60706254/734f5b14-cbb5-455a-a19a-fbe56b920989)
![Screenshot_2](https://github.com/capawesome-team/capacitor-plugins/assets/60706254/45f27ed6-d31e-4349-8b99-49b7d9f4cf86)

Using the display file methods for images it works perfectly, but when using the methods to open files or event the images that I can display it throws the error attached above.

**Comment by robingenz** [comment, marked as answer]
You need to specify the sharable directories correctly, see [Specify sharable directories](https://developer.android.com/training/secure-file-sharing/setup-sharing#DefineMetaData).

**Reply by fabiofilipegomes** [reply]
I'll try, thanks for the fast replay!

**Reply by fabiofilipegomes** [reply]
Hey I've updated the paths but it doesn't work.
Before, the paths in the file_path.xml file were just a ".", and I've updated it to "files/" to match my mediaPath variable. Prints below.

![mediapath](https://github.com/capawesome-team/capacitor-plugins/assets/60706254/ebd3418b-e947-4820-8ea5-be11724b464d)
![paths screenshot](https://github.com/capawesome-team/capacitor-plugins/assets/60706254/653c1928-920e-4004-9678-19d17273ef71)

What should I do now?
Thanks.

**Reply by robingenz** [reply]
Since this is not a bug in the plugin but a misconfiguration, I recommend that you post your question on Stackoverflow. I try to spend as much time as I can on new features and bug fixes.

Also have a look at existing SO answers: https://stackoverflow.com/a/42516202/6731412
